ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/MinimalState.java

```diff
@@ -82,6 +83,38 @@ public class MinimalState {
 
 	protected String fSystemBundle = IPDEBuildConstants.BUNDLE_OSGI;
 
+	private static final IPreferenceChangeListener PREF_CHANGE_LISTENER = e -> {
+		if (e.getKey().equals("org.eclipse.jdt.launching.PREF_DEFAULT_ENVIRONMENTS_XML")) { //$NON-NLS-1$
+			Object oldValue = e.getOldValue() == null ? "" : e.getOldValue(); //$NON-NLS-1$
+			Object newValue = e.getNewValue() == null ? "" : e.getNewValue(); //$NON-NLS-1$
+			if (!oldValue.equals(newValue)) {
+				triggerSystemPackagesReload();
+			}
+		}
+	};
+
+	private static final IVMInstallChangedListener VM_CHANGED_LISTENER = new IVMInstallChangedListener() {
+		@Override
+		public void vmRemoved(IVMInstall vm) {
+			triggerSystemPackagesReload();
+		}
+
+		@Override
+		public void vmChanged(PropertyChangeEvent event) {
+			triggerSystemPackagesReload();
+		}
+
+		@Override
+		public void vmAdded(IVMInstall vm) {
+			triggerSystemPackagesReload();
+		}
+
+		@Override
+		public void defaultVMInstallChanged(IVMInstall previous, IVMInstall current) {
+			triggerSystemPackagesReload();
+		}
+	};
+
```

Comment:
Could you please move this static final fields to the very top of this class to follow the usual convention?